WordPress/gutenberg
# Only display the modified post date if the post has been modified

## What?

The post date block has a setting that displays the modified date. When the setting is enabled, it displays the post date
if the post has not been modified.

This PR updates the block to only display the modified date if the setting is enabled and the post has been modified.
If the setting is enabled, but the post has not been modified, nothing is shown on the front.
It also adds a new CSS class to identify that the setting is enabled so that developers can style it differently.

Fixes [#46645](https://github.com/WordPress/gutenberg/issues/46645)

## Why?

Without this PR, a template that displays both the published date and the modified date could show two identical dates,
and there was no way to hide the modified date if the post was not modified.

## How?

The PR compares the modified date and the post date and returns early if the time difference is a less than a minute.

## Testing Instructions





1. Create a new post.
2. Add a post date block.
3. _Wait a minute or two_ and add a second post date block. Enable the "Display last modified date" setting.
4. Save and view the front. Both blocks should show, with different dates.

5. Create a second new post.
6. Add a post date block. Enable the "Display last modified date" setting.
7. Save and view the front. If you were fast enough, the block should not show.
8. Go back to the editor, _wait a minute_ and make a change to the post.
9. Save and view the front, confirm that the block shows.

### Testing Instructions for Keyboard


## Screenshots or screencast
